loader/icd.c

```diff
@@ -132,27 +144,77 @@ void khrIcdVendorAdd(const char *libraryName)
         char *suffix;
         size_t suffixSize;
 
-        // call clGetPlatformInfo on the returned platform to get the suffix
-        if (!platforms[i])
+        // skip NULL platforms and non dispatchable platforms
+        if (!platforms[i] || !platforms[i]->dispatch)
```

Comment:
How can dispatch be NULL?

Reply:
I just extended the original check, to ensure that the returned platform is valid. The idea is not to register platform that obviously do not match the OpenCL icd constraints.